addons/medical/scripts/Game/ACE_Medical/Character/SCR_CharacterControllerComponent.c

```diff
@@ -5,54 +5,77 @@ modded class SCR_CharacterControllerComponent : CharacterControllerComponent
 {
 	protected SCR_CharacterDamageManagerComponent m_pACE_Medical_DamageManager;
 	protected const float ACE_MEDICAL_SECOND_CHANCE_DEACTIVATION_TIMEOUT_MS = 1000;
-	
+
 	//------------------------------------------------------------------------------------------------
-	//! Initialize member variables
+	//! Initialize member variables.
 	override void OnInit(IEntity owner)
 	{
 		super.OnInit(owner);
 		m_pACE_Medical_DamageManager = SCR_CharacterDamageManagerComponent.Cast(owner.FindComponent(SCR_CharacterDamageManagerComponent));
+
+		// Add null check to avoid potential null reference errors
+		if (!m_pACE_Medical_DamageManager)
+		{
+			Print("ACE Medical Damage Manager not found!", LogLevel.ERROR);
+			return;
+		}
 	}
-	
+
 	//------------------------------------------------------------------------------------------------
 	//! Add/remove second chance when life state changes
 	override void OnLifeStateChanged(ECharacterLifeState previousLifeState, ECharacterLifeState newLifeState)
 	{
 		super.OnLifeStateChanged(previousLifeState, newLifeState);
-		
-		// Only run if ACE Medical has been initialized for this character
+
+		// Check if ACE Medical has been initialized
 		if (!m_pACE_Medical_DamageManager.ACE_Medical_IsInitialized())
 			return;
-		
+
 		// OnLifeStateChanged sometimes gets triggered without a change in state
 		if (previousLifeState == newLifeState)
 			return;
-		
+
 		switch (newLifeState)
 		{
-			// Add second chance when revived
 			case ECharacterLifeState.ALIVE:
-			{
-				GetGame().GetCallqueue().Remove(m_pACE_Medical_DamageManager.ACE_Medical_EnableSecondChance);
-				m_pACE_Medical_DamageManager.ACE_Medical_EnableSecondChance(true);
-				m_pACE_Medical_DamageManager.ACE_Medical_SetSecondChanceTrigged(false);
+				HandleAliveState();
 				break;
-			}
-			
-			// Schedule removal of second chance when falling unconscious
+
 			case ECharacterLifeState.INCAPACITATED:
-			{
-				GetGame().GetCallqueue().CallLater(m_pACE_Medical_DamageManager.ACE_Medical_EnableSecondChance, ACE_MEDICAL_SECOND_CHANCE_DEACTIVATION_TIMEOUT_MS, false, false);
+				HandleIncapacitatedState();
 				break;
-			}
-			
-			// Remove second chance when dead
+
 			case ECharacterLifeState.DEAD:
-			{
-				GetGame().GetCallqueue().Remove(m_pACE_Medical_DamageManager.ACE_Medical_EnableSecondChance);
-				m_pACE_Medical_DamageManager.ACE_Medical_EnableSecondChance(false);
+				HandleDeadState();
 				break;
-			}
 		}
 	}
+
+	//------------------------------------------------------------------------------------------------
+	//! Handle logic when character is revived (ALIVE state)
+	void HandleAliveState()
```

Comment:
this is not a public API
```suggestion
	protected void HandleAliveState()
```